### PR TITLE
Refactor unit tests for vector functions.

### PR DESCRIPTION
**Pull request by jtibshirani**
This PR performs the following changes:
* Split `ScoreScriptUtilsTests` into `DenseVectorFunctionTests` and
`SparseVectorFunctionTests`. This will make it easier to delete all sparse
vector function tests once we remove support on 8.x.
* As much as possible, break up the large test methods into individual tests
for each vector function (`cosineSimilarity`, `l2norm`, etc.).

### CODE REVIEW

**Review by idjevm — APPROVED** [review]

File: x-pack/plugin/vectors/src/test/java/org/elasticsearch/xpack/vectors/query/DenseVectorFunctionTests.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    @Before
+    public void setUpVectors() {
+        field = "vector";
+        docVector = new float[] {230.0f, 300.33f, -34.8988f, 15.555f, -200.0f};
+        queryVector = Arrays.asList(0.5f, 111.3f, -13.0f, 14.8f, -156.0f);
+        invalidQueryVector = Arrays.asList(0.5, 111.3);
+    }
+
+    public void testDenseVectorFunctions() {
+        for (Version indexVersion : Arrays.asList(Version.V_7_4_0, Version.CURRENT)) {
```
Comment (mayya-sharipova) [inline x-pack/plugin/vectors/src/test/java/org/elasticsearch/xpack/vectors/query/DenseVectorFunctionTests.java]: We have a loop here as we expect a change in every version?

**Review by mayya-sharipova — APPROVED** [review]
@jtibshirani Thank you, very nice refactoring. I like how the tests are so clearly organized, and very fast to  understand. I will adopt your style :)

File: x-pack/plugin/vectors/src/test/java/org/elasticsearch/xpack/vectors/query/DenseVectorFunctionTests.java (hunk 1 above)
Comment (jtibshirani) [reply x-pack/plugin/vectors/src/test/java/org/elasticsearch/xpack/vectors/query/DenseVectorFunctionTests.java]: I wasn't planning to add more versions (unless we have more version-dependent changes). The loop is really just to test these two listed versions.

**Comment by jtibshirani** [comment]
Thanks @mayya-sharipova for the review (and @idjevm for taking a look as well!)

**Pull request merged**